### PR TITLE
Fix for Issue 1992

### PR DESCRIPTION
**Pull request by qianfengz**
This P.R is for fixing the left issues in several previous P.Rs mentioned [Issue 1992](https://github.com/ROCmSoftwarePlatform/MIOpen/issues/1992)

### CODE REVIEW

File: src/solver/conv_asm_implicit_gemm_gtc_fwd.cpp [hunk 1]
```diff
@@ -1534,7 +1534,8 @@ bool ConvAsmImplicitGemmGTCDynamicFwdXdlops::IsApplicable(const ExecutionContext
 
 #if WORKAROUND_SWDEV_306318
     if((problem.kernel_size_h == 1) && (problem.kernel_size_w == 1) && (problem.n_inputs % 8 != 0))
-        return false;
+        if(!miopen::IsEnabled(MIOPEN_DEBUG_CONV_IMPLICIT_GEMM_ASM_FWD_GTC_XDLOPS{}))
+            return false;
```
Comment (carlushuang) [inline src/solver/conv_asm_implicit_gemm_gtc_fwd.cpp:1538]: image
we have checked this `MIOPEN_DEBUG_CONV_IMPLICIT_GEMM_ASM_FWD_GTC_XDLOPS` when entering this function, is there any special reason need to check it again here?

**Review by carlushuang — DISMISSED** [review]
LGTM

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -1342,27 +1342,28 @@ add_custom_test(test_conv_igemm_dynamic_xdlops_bwd_float SKIP_UNLESS_ALL HALF_DI
 COMMAND ${DYNAMIC_IMPLICITGEMM_BWD_ENVS_XDLOPS} $<TARGET_FILE:test_conv2d> ${MIOPEN_TEST_FLOAT_ARG} --verbose --input  4  512 128 128 --weights 12  512  1 1 --pads_strides_dilations 0 0 1 1 1 1 --disable-forward --disable-backward-weights
 )
 
-#add_custom_test(test_conv_igemm_dynamic_xdlops_fwd SKIP_UNLESS_ALL HALF_ENABLED GFX90A_DISABLED GFX900_DISABLED GFX906_DISABLED SKIP_XNACK_ON
+# Commented out due to WORKAROUND_SWDEV_306318
```
Comment (atamazov) [inline test/CMakeLists.txt]: AFAICS `C % 8 == 0` in all these tests, so this W/A in tests is not needed at all, right?

**Review by atamazov — CHANGES_REQUESTED** [review]

File: test/CMakeLists.txt (hunk 2 above)
Comment (atamazov) [reply test/CMakeLists.txt]: I mean we can enable all test cases.

Comment (qianfengz) [reply test/CMakeLists.txt]: @atamazov  I see, thank you.  Updated

**Review by atamazov — APPROVED** [review]
Looks good!

**Review by junliume — APPROVED** [review]

**Pull request merged**